File: Daily Questions/0860-lemonade-change/0860-lemonade-change.c
bool lemonadeChange(int* bills, int billsSize) {
    int fiveBillsCount = 0, tenBillsCount = 0;
    for (int i = 0; i < billsSize; i++) {
        if (bills[i] == 5) {
            fiveBillsCount++;
        }
        else if (bills[i] == 10) {
            tenBillsCount++;
            fiveBillsCount--;
        }
        else {
            if (tenBillsCount > 0) {
                tenBillsCount--;
                fiveBillsCount--;
            }
            else {
                fiveBillsCount -= 3;
            }
        }
        if (fiveBillsCount < 0) {
            return false;
        }
    }
    return true;
}

File: Difficulty/A. Easy/0860-lemonade-change/0860-lemonade-change.py
class Solution:
    def lemonadeChange(self, bills: List[int]) -> bool:
        changeList = [0, 0]
        for bill in bills:
            if bill == 5:
                changeList[0] += 1
            elif bill == 10:
                changeList[1] += 1
                changeList[0] -= 1
            else:
                if changeList[1] > 0:
                    changeList[1] -= 1
                    changeList[0] -= 1
                else:
                    changeList[0] -= 3
            if changeList[0] < 0:
                return False
        return True

        